Tests/Shared.Specs/CollectionEquivalencySpecs.cs

```diff
@@ -182,6 +183,34 @@ public void Add(Guid userId, params string[] roles)
             }
         }
 
+        [Fact]
+        public void When_BeEquivalentTo_calls_GetEqualityStrategy_it_should_be_thread_safe()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            IEnumerable collection1 = new[] { 1 };
+            IEnumerable collection2 = new[] { 1 };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => Parallel.For(
+                0,
+                10_000,
+                new ParallelOptions
+                {
+                    MaxDegreeOfParallelism = 8
+                },
+                e => collection1.Should().BeEquivalentTo(collection2)
+            );
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
```

Comment:
🤔 will `Parallel.For` ever throw?

Reply:
Yes. It will wrap the exceptions in an `AggregateException`.
We did a similar test when testing the thread-safety for `Configuration`.
https://github.com/fluentassertions/fluentassertions/blob/master/Tests/Shared.Specs/ConfigurationSpecs.cs